### PR TITLE
fix: debian apt-get update running before configuring repositories

### PR DESCRIPTION
**Pull request by cnvergence**
**What this PR does / why we need it**:
In debian script we were running apt-get update before configuring new repositories for Kubernetes.
When upgrading the KubeOne cluster before these changes, it can lead to an error with trying to download Kubernetes release from deprecated packages.cloud.google.com.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/kubeone/issues/3077

**What type of PR is this?**

/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Fixed Kubeone upgrades on Debian hosts with deprecated Kubernetes repositories when running apt-get update.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: pkg/scripts/os_debian.go [hunk 1]
```diff
@@ -33,6 +33,14 @@ source /etc/kubeone/proxy-env
 {{ template "sysctl-k8s" . }}
 {{ template "journald-config" }}
 
+{{- if .CONFIGURE_REPOSITORIES }}
+sudo install -m 0755 -d /etc/apt/keyrings
+
+curl -fsSL https://pkgs.k8s.io/core:/stable:/{{ .KUBERNETES_MAJOR_MINOR }}/deb/Release.key | sudo gpg --dearmor --yes -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
```
[the remark this answers is not in the document]
Comment (cnvergence) [reply pkg/scripts/os_debian.go:39]: ah right, it's using a curl

**Review by xmudrii — APPROVED** [review]
Thank you, @cnvergence!
/lgtm
/approve

**Pull request merged**